server/app/controllers/obslist.py

```diff
@@ -1,17 +1,56 @@
+from datetime import timedelta
 from app import app
 from app.repository import Repository
 from app.pagination import use_pagination
 
+from webargs import fields
+from webargs.flaskparser import use_kwargs
+
 @app.route('/obslist')
 @use_pagination()
-def obslist(limit_and_offset):
-    '''This function retrieves observations list from a local database and displays it.'''
+# Use_kwargs introduces ugly behavior when user provide parameter without value,
+# for example "?sat_id=". It should be ignored or handle as None value, but
+# I cannot set them. Now user get HTTP 422 Unprocessable Entity. I think that
+# it isn't important, because user doesn't provide it manually.
+@use_kwargs({
+    'aos_before': fields.Date(),
+    'los_after': fields.Date(),
+    'sat_id': fields.Int(),
+    'station_id': fields.Int(),
+    'has_tle': fields.Bool()
+}, validate=lambda kwargs: "aos_before" not in kwargs or
```

Comment:
What happens if the fields values fail validation?

Reply:
You get HTTP 442 Unprocessable entity (by default, customizable).
User never see this status page, because we have also client-side validation.